scripts/_helpers.py

```diff
@@ -150,3 +151,88 @@ def dlProgress(count, blockSize, totalSize):
 
     urllib.request.urlretrieve(url, file, reporthook=dlProgress)
 
+
+def mocksnakemake(rulename, **wildcards):
+    """
+    This function is expected to be executed in the 'scripts'-directory within
+    the snakemake project. It returns a snakemake.script.Snakemake object,
+    based on the Snakefile of the upper directory, such that includes
+    mostly all necessary information (input, output, log etc.).
+
+    If a rule has wildcards, you have to specify them in **wildcards.
+
+    Parameters
+    ----------
+    rulename: str
+        name of the rule for which the snakemake object should be generated
+    **wildcards:
+        keyword arguments fixing the wildcards. Only necessary if wildcards are
+        needed.
+    """
+    import snakemake as sm
+    import os
+    import logging
+    from pypsa.descriptors import Dict
+    from os.path import abspath
+
+    os.chdir('..')
+    for p in sm.SNAKEFILE_CHOICES:
+        if os.path.exists(p):
+            snakefile = p
+            break
+    workflow = sm.Workflow(snakefile)
+    workflow.include(snakefile)
+    rule = workflow.get_rule(rulename)
+    wc = Dict(wildcards)
+
+    # make the input files accessable by taking the absolut paths
+    def make_io_accessable(smfiles):
+        if not smfiles:
+            return
+        # for mildy hacky input functions (like make_summary):
+        if len(smfiles) == 1 and callable(smfiles[0]):
+            new_input_files = smfiles[0](wc)
+            # overwrite smfiles with extracted files from function
+            smfiles = sm.io.InputFiles()
+            for index, p in enumerate(new_input_files):
+                smfiles.insert(index, p)
+        # now iterate over each item and make path an absolut path
+        files = sm.io.InputFiles()
+        for index, (key, p) in enumerate(smfiles.allitems()):
+            # case that item is a function
+            if callable(p):
+                p = p(wc)
+            if isinstance(p, str):
+                files.insert(index, sm.io.apply_wildcards(abspath(p), wc))
+            else:
+                files.insert(index, p)
+            if key is not None:
+                files.add_name(key)
+        return files
+
+    Input = make_io_accessable(rule.input)
+    Output = make_io_accessable(rule.output)
+    if not rule.log:
+        rule.log.insert(0, f"logs/{rule.name}.log")
```

Comment:
```suggestion
```

Since you only want to mock the `snakemake` object, I would not include additional assumptions in here.
Especially for debugging, additions to the `snakemake` object here will make it more complicated.